bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/classic/BigtableHBaseClassicSettings.java

```diff
@@ -0,0 +1,348 @@
+      bigtableOptionsBuilder.setAdminHost(adminHostOverride);
+    }
+
+    String portOverrideStr = configuration.get(BigtableOptionsFactory.BIGTABLE_PORT_KEY);
+    if (portOverrideStr != null) {
+      bigtableOptionsBuilder.setPort(Integer.parseInt(portOverrideStr));
+    }
+
+    String usePlaintextStr = configuration.get(BIGTABLE_USE_PLAINTEXT_NEGOTIATION);
+    if (usePlaintextStr != null) {
+      bigtableOptionsBuilder.setUsePlaintextNegotiation(Boolean.parseBoolean(usePlaintextStr));
+    }
+
+    setBulkOptions(bigtableOptionsBuilder);
+    setChannelOptions(bigtableOptionsBuilder);
+    setClientCallOptions(bigtableOptionsBuilder);
+
+    String emulatorHost = configuration.get(BigtableOptionsFactory.BIGTABLE_EMULATOR_HOST_KEY);
+    if (emulatorHost != null) {
+      bigtableOptionsBuilder.enableEmulator(emulatorHost);
+    }
+
+    String useBatchStr = configuration.get(BIGTABLE_USE_BATCH);
+    if (useBatchStr != null) {
+      bigtableOptionsBuilder.setUseBatch(Boolean.parseBoolean(useBatchStr));
+    }
+
+    String useGcjClientStr = configuration.get(BIGTABLE_USE_GCJ_CLIENT);
+    if (useGcjClientStr != null) {
+      bigtableOptionsBuilder.setUseGCJClient(Boolean.parseBoolean(useGcjClientStr));
+    }
+    return bigtableOptionsBuilder.build();
+  }
+  // </editor-fold>
+
+  // <editor-fold desc="Private Helpers">
+  private void setChannelOptions(BigtableOptions.Builder builder) throws IOException {
+    setCredentialOptions(builder);
+
+    builder.setRetryOptions(createRetryOptions());
+
+    String channelCountStr = configuration.get(BIGTABLE_DATA_CHANNEL_COUNT_KEY);
+    if (channelCountStr != null) {
+      builder.setDataChannelCount(Integer.parseInt(channelCountStr));
+    }
+
+    // This is primarily used by Dataflow where connections open and close often. This is a
+    // performance optimization that will reduce the cost to open connections.
+    String useCachedDataPoolStr = configuration.get(BIGTABLE_USE_CACHED_DATA_CHANNEL_POOL);
+    if (useCachedDataPoolStr != null) {
+      builder.setUseCachedDataPool(Boolean.parseBoolean(useCachedDataPoolStr));
+    }
+
+    // This information is in addition to bigtable-client-core version, and jdk version.
+    StringBuilder agentBuilder = new StringBuilder();
+    agentBuilder.append("hbase-").append(VersionInfo.getVersion());
+    String customUserAgent = configuration.get(CUSTOM_USER_AGENT_KEY);
+    if (customUserAgent != null) {
+      agentBuilder.append(',').append(customUserAgent);
+    }
+    builder.setUserAgent(agentBuilder.toString());
+  }
+
+  private void setBulkOptions(BigtableOptions.Builder bigtableOptionsBuilder) {
+    BulkOptions.Builder bulkOptionsBuilder = BulkOptions.builder();
+
+    int asyncMutatorCount =
+        configuration.getInt(
+            BIGTABLE_ASYNC_MUTATOR_COUNT_KEY, BIGTABLE_ASYNC_MUTATOR_COUNT_DEFAULT);
+    bulkOptionsBuilder.setAsyncMutatorWorkerCount(asyncMutatorCount);
```

Comment:
Although it's not as important as in the hostname case, I think you should avoid double defining the defaults and follow the pattern:
```
value =  config.get("key")
if (value != null) {
  options.setValue(value);
}
```

Same for everything below.